core/src/main/java/org/apache/iceberg/avro/AvroCustomOrderSchemaVisitor.java

```diff
@@ -37,6 +37,10 @@ public static <T, F> T visit(Schema schema, AvroCustomOrderSchemaVisitor<T, F> v
 
         visitor.recordLevels.push(name);
 
+        if (schema.getLogicalType() instanceof Variant) {
+          return visitor.variant(schema);
```

Comment:
Similar to my comment on `AvroSchemaVisitor`, I think this should visit the `metadata` and `value` children.